src/DatabaseUpdate/components/checkUpdate.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Animated, Text, View, Easing } from "react-native";
+import { Icon, ListItem } from "@rneui/themed";
+import { STRINGS } from "@common";
+import PropTypes from "prop-types";
+import { useSelector } from "react-redux";
+import { checkUpdateStyles as styles, darkMode } from "./styles";
+
+const CheckUpdatesAnimation = ({ isLoading, isUpdateAvailable }) => {
+  const isNightMode = useSelector((state) => state.isNightMode);
+  const rotateAnim = useRef(new Animated.Value(0)).current;
+  const [checking, setChecking] = useState(false);
+  const animationRef = useRef(null);
+  const { darkModeContainer, darkModeText } = darkMode(isNightMode);
+
+  useEffect(() => {
+    if (checking) {
+      // Reset the animated value to 0 before starting
+      rotateAnim.setValue(0);
+      animationRef.current = Animated.loop(
+        Animated.timing(rotateAnim, {
+          toValue: 1,
+          duration: 1000,
+          easing: Easing.linear,
+          useNativeDriver: true,
+        })
+      );
+      animationRef.current.start();
+    } else if (animationRef.current) {
+      animationRef.current.stop();
+      rotateAnim.setValue(0); // Reset after stopping
+    }
+
+    // Clean up on unmount
+    return () => {
+      if (animationRef.current) {
+        animationRef.current.stop();
+      }
+    };
+  }, [checking, rotateAnim]);
+
+  const handleCheckUpdates = () => {
+    setChecking(true);
+    // Simulate an update check with a timeout of 3 seconds
+    setTimeout(() => {
```

Comment:
Store the timeout ID and clear it on unmount to prevent potential memory leaks if the component unmounts before the timeout finishes.